src/components/Carousel.tsx

```diff
@@ -1,18 +1,125 @@
 import React from 'react';
 import './Carousel.scss';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+type State = {
+  position: number,
+};
+
+type Props = {
+  images: string[],
+  step: number,
+  frameSize: number,
+  itemWidth: number,
+  animationDuration: number,
+  infinite: boolean,
+};
+
+class Carousel extends React.Component<Props, State> {
+  state = {
+    position: 0,
+  };
+
+  moveSlider = (buttonName: string) => {
```

Comment:
you can add specific typing for `buttonName`:
```
enum ControlButton {
  Prev,
  Next,
}
```
and use it in `moveSlider` function
```suggestion
  moveSlider = (buttonName: ControlButton) => {
```